File: bookstore-backend/src/main/java/com/bookstore/entity/Payment.java
package com.bookstore.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false, unique = true)
    @NotNull(message = "Order is required")
    private Order order;

    @Column(name = "payment_method", nullable = false)
    @Enumerated(EnumType.STRING)
    private PaymentMethod paymentMethod;

    @Column(name = "amount", nullable = false, precision = 10, scale = 2)
    @NotNull(message = "Amount is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Amount must be greater than 0")
    private BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentStatus status = PaymentStatus.PENDING;

    @Column(name = "transaction_id", unique = true)
    private String transactionId;

    @Column(name = "payment_gateway")
    private String paymentGateway;

    @Column(name = "gateway_response", columnDefinition = "TEXT")
    private String gatewayResponse;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "paid_at")
    private LocalDateTime paidAt;

    @Column(name = "refunded_at")
    private LocalDateTime refundedAt;

    @Column(name = "refund_amount", precision = 10, scale = 2)
    private BigDecimal refundAmount;

    @Column(name = "refund_reason")
    private String refundReason;

    public enum PaymentMethod {
        ALIPAY("支付宝"),
        WECHAT_PAY("微信支付"),
        BANK_CARD("银行卡"),
        CREDIT_CARD("信用卡");

        private final String displayName;

        PaymentMethod(String displayName) {
            this.displayName = displayName;
        }

        public String getDisplayName() {
            return displayName;
        }
    }

    public enum PaymentStatus {
        PENDING("待支付"),
        PROCESSING("支付中"),
        SUCCESS("支付成功"),
        FAILED("支付失败"),
        CANCELLED("已取消"),
        REFUNDED("已退款"),
        PARTIAL_REFUNDED("部分退款");

        private final String displayName;

        PaymentStatus(String displayName) {
            this.displayName = displayName;
        }

        public String getDisplayName() {
            return displayName;
        }
    }
}

File: bookstore-frontend/src/types/payment.ts
export interface Payment {
  id: number;
  orderId: number;
  paymentMethod: PaymentMethod;
  paymentMethodName: string;
  amount: number;
  status: PaymentStatus;
  statusName: string;
  transactionId: string;
  paymentGateway: string;
  createdAt: string;
  paidAt?: string;
  refundedAt?: string;
  refundAmount?: number;
  refundReason?: string;
  paymentUrl?: string;
  qrCodeData?: string;
}

export type PaymentResponse = Payment;

export enum PaymentMethod {
  ALIPAY = 'ALIPAY',
  WECHAT_PAY = 'WECHAT_PAY',
  BANK_CARD = 'BANK_CARD',
  CREDIT_CARD = 'CREDIT_CARD'
}

export enum PaymentStatus {
  PENDING = 'PENDING',
  PROCESSING = 'PROCESSING',
  SUCCESS = 'SUCCESS',
  FAILED = 'FAILED',
  CANCELLED = 'CANCELLED',
  REFUNDED = 'REFUNDED',
  PARTIAL_REFUNDED = 'PARTIAL_REFUNDED'
}

export interface PaymentRequest {
  orderId: number;
  paymentMethod: PaymentMethod;
  returnUrl?: string;
  notifyUrl?: string;
}

export interface RefundRequest {
  paymentId: number;
  refundAmount: number;
  refundReason: string;
  adminNote?: string;
}

export interface PaymentStatistics {
  totalSuccessfulPayments: number;
  totalPaymentAmount: number;
  totalRefundAmount: number;
  statusStatistics: Record<string, number>;
}

export const PaymentMethodLabels: Record<PaymentMethod, string> = {
  [PaymentMethod.ALIPAY]: '支付宝',
  [PaymentMethod.WECHAT_PAY]: '微信支付',
  [PaymentMethod.BANK_CARD]: '银行卡',
  [PaymentMethod.CREDIT_CARD]: '信用卡'
};

export const PaymentStatusLabels: Record<PaymentStatus, string> = {
  [PaymentStatus.PENDING]: '待支付',
  [PaymentStatus.PROCESSING]: '支付中',
  [PaymentStatus.SUCCESS]: '支付成功',
  [PaymentStatus.FAILED]: '支付失败',
  [PaymentStatus.CANCELLED]: '已取消',
  [PaymentStatus.REFUNDED]: '已退款',
  [PaymentStatus.PARTIAL_REFUNDED]: '部分退款'
};